### PR TITLE
feat: add route for openrice.com

### PR DESCRIPTION
**Pull request by after9**
## Involved Issue / 该 PR 相关 Issue

NOROUTE

## Example for the Proposed Route(s) / 路由地址示例

```routes
/openrice/zh/hongkong/promos
/openrice/zh/hongkong/offers
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [ ] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明

Openrice.com是立足香港，辐射亚洲多国的美食信息网站。这次主要添加了香港本土的美食信息以及实时优惠信息。后期会增加餐厅排名及其它国家地区的美食资讯。

### CODE REVIEW

File: lib/routes/openrice/offers.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        },
+    });
+    const highlightedOffers = response.highlightedOffers;
+    const normalOffers = response.searchResult.paginationResult.results;
+    const data = [...highlightedOffers, ...normalOffers];
+
+    const resultList = data.map((item) => {
+        const title = item.title ?? '';
+        const link = baseUrl + item.urlUI;
+        const coverImg = item.doorPhotoUI.url ?? '';
```
Comment (TonyRL) [inline lib/routes/openrice/offers.ts:67]:
```suggestion
        const coverImg = item.doorPhotoUI.urls.full ?? '';
```

File: lib/routes/openrice/templates/description.art [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{{@ description }}
```
Comment (TonyRL) [inline lib/routes/openrice/templates/description.art:1]:
```suggestion
{{ description }}
```
You don't need raw mode if `description` does not contain any HTML code

File: lib/routes/openrice/promos.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        case 'en':
+            title = "Openrice - What's Hot";
+            description = 'Hot Restaurant in Hong Kong from Openrice';
+            urlPath = '/en/hongkong/promos';
+            break;
+        default:
+            title = '開飯喇 - 餐厅滋讯';
+            description = 'OpenRice香港餐廳滋訊';
+            urlPath = '/zh/hongkong/promos';
+            break;
```
Comment (TonyRL) [inline lib/routes/openrice/promos.ts:51]:
```suggestion
        case 'zh-cn':
            title = '开饭喇 - 餐厅资讯';
            description = 'OpenRice香港餐厅资讯';
            urlPath = '/zh-cn/hongkong/promos';
            break;
        case 'en':
            title = "Openrice - What's Hot";
            description = 'Hot Restaurant in Hong Kong from Openrice';
            urlPath = '/en/hongkong/promos';
            break;
        case 'zh':
        default:
            title = '開飯喇 - 餐厅滋讯';
            description = 'OpenRice香港餐廳滋訊';
            urlPath = '/zh/hongkong/promos';
            break;
```

File: lib/routes/openrice/offers.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+import { getCurrentPath } from '@/utils/helpers';
+const __dirname = getCurrentPath(import.meta.url);
+const baseUrl = 'https://www.openrice.com';
+
+export const route: Route = {
+    path: '/:lang/hongkong/offers',
+    maintainers: ['after9'],
+    handler,
+    categories: ['shopping'],
+    example: '/openrice/zh/hongkong/promos',
```
Comment (TonyRL) [inline lib/routes/openrice/offers.ts:14]:
```suggestion
    example: '/openrice/zh/hongkong/offers',
```

**Comment by after9** [comment]
Hi @TonyRL , the suggestions are quite valueable for me. I had udpate the code as your suggestions. Please check again.
Thanks!

**Review by TonyRL — COMMENTED** [review]
https://deepscan.io/dashboard/#view=project&tid=6244&pid=8135&bid=92448&subview=pull-request&prid=2018867

**Pull request merged**